lib/ansible/utils/collection_loader.py

```diff
@@ -550,9 +550,9 @@ def get_collection_name_from_path(path):
     :param n_path: native-string path to evaluate for collection containment
     :return: collection name or None
     """
-    n_collection_paths = [to_native(os.path.realpath(to_bytes(p))) for p in AnsibleCollectionLoader().n_collection_paths]
+    n_collection_paths = [to_native(os.path.abspath(to_bytes(p))) for p in AnsibleCollectionLoader().n_collection_paths]
```

Comment:
Why the change here and below from `realpath` to `abspath`?

Reply:
to allow for symlinks

Reply:
Is that related to the ansible-doc fixes? It seems like a separate issue.

Reply:
related to testing them, i  had complaint docs didn't work for several people, while loading the collection plugins did work, i traced it back to them using symlinks.

Reply:
Could you put those fixes in a separate PR and include tests with it to prevent regressions?